src/odemis/driver/test/smaract_test.py

```diff
@@ -649,6 +652,56 @@ def test_auto_update_function(self):
         pos_after_move = self.dev.position.value
         testing.assert_pos_almost_equal(expected_pos, pos_after_move, **COMP_ARGS)
 
+    def test_param_file_nofile(self):
+        # check if forcing a ValueError is the result of a file that does not exist
+        with self.assertRaises(ValueError):
+            CONFIG_3DOF['param_file'] = "nonexistingfile.mcs2.tsv"
+            smaract.MCS2(**CONFIG_3DOF)
+
+    def test_parse_tsv_config(self):
+        # create a temporary .mcs2.tsv file
+        file_id, file_dir = tempfile.mkstemp(suffix='mcs2.tsv', text=True)
+
+        # open the file descriptor/id as a regular file in write mode
+        testfile = os.fdopen(file_id, 'w')
+
+        # create a dictionary with parameter identifiers and values
+        param_dict = [{"num": 1, "code": 0x0307005D, "value": 0b00000001, "type": "I32"},
+                      {"num": 1, "code": 0x0309005D, "value": 0b00001000, "type": "I64"},
+                      {"num": 1, "code": 0x0308007E, "value": "Newname", "type": "Str"}]
+
+        # add a comment line to test for
+        testfile.writelines("# commented line example\n")
+        # add an int32 type property line to test for
+        testfile.writelines(f"{param_dict[0]['num']}\t{hex(param_dict[0]['code'])}\t{param_dict[0]['value']}\t"
+                            f"{param_dict[0]['type']} # Referencing Options (Inverted Start Direction)\n")
+        # add an int64 type property line to test for
+        testfile.writelines(f"{param_dict[1]['num']}\t{hex(param_dict[1]['code'])}\t{param_dict[1]['value']}\t"
+                            f"{param_dict[1]['type']} # Referencing Options (Inverted Start Direction)\n")
+        # add a String type property line to test for
+        testfile.writelines(f"{param_dict[2]['num']}\t{hex(param_dict[2]['code'])}\t{param_dict[2]['value']}\t"
+                            f"{param_dict[2]['type']} # Referencing Options (Inverted Start Direction)\n")
+
+        CONFIG_3DOF['param_file'] = file_dir
+        dev = smaract.MCS2(**CONFIG_3DOF)
+
+        # open the target file again, this time for reading only
+        testfile = open(file_dir, 'r')
+        out_params = dev.parse_tsv_config(testfile)
+
+        # check if the output dict contains the right amount of properties
+        self.assertEqual(len(out_params), len(param_dict))
+
+        # check if the output dict contains the right property input values
+        for item in param_dict:
+            key = (item['num'], item['code'], item['type'])
+            if key in out_params:
+                self.assertEqual(item['value'], out_params[key])
+
+        testfile.close()
+        os.unlink(file_dir)
```

Comment:
Please just use `os.remove()`, as it's easier to understand what it does. Or does it do something different?

Reply:
I think it does not do anything different.. unlink was more of a Linux way to remove files. But both can be used interchangeably.